devops/sftp/program.py

```diff
@@ -0,0 +1,59 @@
+"""Download a file via SFTP, remove spaces and upload it."""
+
+from io import BytesIO
+import os
+
+import autokitteh
+import paramiko
+import requests
+
+
+SFTP_CONFIG = {
+    "host": "test.rebex.net",
+    "port": 22,
+    "username": "demo",
+    "password": "password",
+    "remote_path": "/readme.txt",
+}
+
+
+HTTP_TARGET = os.getenv("HTTP_TARGET")
+
+
+def download_sftp_file(host, port, username, password, remote_path):
+    transport = paramiko.Transport((host, port))
+    transport.connect(username=username, password=password)
+    sftp = paramiko.SFTPClient.from_transport(transport)
+
+    file_io = BytesIO()
+    sftp.getfo(remote_path, file_io)
+    file_io.seek(0)
+
+    sftp.close()
+    transport.close()
+    return file_io
+
+
+def remove_spaces(file_io):
+    content = file_io.read().decode("utf-8").replace(" ", "")
+    return BytesIO(content.encode("utf-8"))
+
+
+def send_file_http(file_io, filename, url):
+    file_io.seek(0)
+    files = {"file": (filename, file_io)}
+    response = requests.post(url, files=files, timeout=10)
+    return response.status_code, response.text
+
+
+@autokitteh.activity
```

Comment:
document while you're using activity here:
- big files boundary
- passing a stream around which is not serializable